src/api/File.js

```diff
@@ -63,6 +63,29 @@ class File extends Item {
             .catch(errorCallback);
     }
 
+    /**
+     * API for setting the description of a file
+     *
+     * @param {string} id - file id
+     * @return {void}
+     */
+    setFileDescription(id: string, value: string, successCallback: Function, errorCallback: Function): Promise<void> {
+        const body = {
+            description: value
+        };
+
+        return this.xhr
+            .put({
+                id: this.getTypedFileId(id),
+                url: this.getUrl(id),
+                data: body
+            })
+            .then((data: BoxItem) => {
+                successCallback(data[FIELD_DOWNLOAD_URL]);
```

Comment:
download url?

Reply:
Also, file in the cache needs to be updated with new changes.
See how things are done here https://github.com/box/box-ui-elements/blob/master/src/api/Item.js

Reply:
Haven't had the need thus far outside preview, but there may be a possibility that you may have to modify the merge function of cache to create a new object instead of just mutating the existing one with new changed data, so that when you do setState() in react, it notices the file change.

Reply:
Is it better to do a permissions check and short circuit vs. attempt the update and handle the failure? To pass in the whole file, I would have to do something like mutate it in the component

`file.description = newValue`

and then pass in that mutated object to the API layer

Reply:
Do permission check and just return for now and not make api call